lib/pbench/server/database/models/datasets.py

```diff
@@ -286,8 +287,11 @@ class Dataset(Database.Base):
     # Dataset name
     name = Column(String(1024), unique=False, nullable=False)
 
-    # OIDC UUID of the owning user
-    owner_id = Column(String(255), nullable=False)
+    # ID of the owning user
+    owner_id = Column(Integer, ForeignKey("users.id"), nullable=False)
+
+    # Indirect reference to the owning User record
+    owner = relationship("User")
```

Comment:
From what I understand need we a relationship because we need to be able to perform Joins on both tables. Without the relationship, our ORM wouldn't be able to join with another table. I think Foreign Key is a SQL operation and relationship is an ORM operation so we need both. Also its a convention we follow in other places too like Dave mentioned [here](https://github.com/distributed-system-analysis/pbench/pull/3232#discussion_r1099077383)

Reply:
I think our requirements are these:
- Given a dataset, we need to be able to determine what the owner's username is.
- Given a username, we need to be able to generate a list of datasets whose owner has that username.

These don't _necessarily_ require joined tables, although that might be an expedient implementation.

For instance, in the first case, when we return a list of (e.g., publicly-available) datasets, it might be a pain to have to translate each owner UUID to its username, so having joined tables might be a substantial boon.

However, in the second case, it might be just as effective to translate the username into one or more UUIDs and then select the datasets based the UUIDs rather than first joining the tables and then selecting based on username.  (Consider that we will likely have a large number of datasets, and potentially a large number of users, and so selecting datasets by user will result in only a fraction of the overall list, which means we would like to avoid doing anything with the datasets that we are _not_ going to return, including joining rows from other tables to them....)

So, it's not obvious to me that we need to be able to perform both joins.

Reply:
> However, in the second case, it might be just as effective to translate the username into one or more UUIDs and then select the datasets based the UUIDs rather than first joining the tables and then selecting based on username.

This is a matter of convenience. One way may not be more effective than the other (In terms of SQL execution the join is performed using the foreign key anyway) but it gives us the flexibility of keeping the collection of related objects on the parent together. It's a SQLAlchemy artifact and it wouldn't hurt to keep them.

Reply:
> This is a matter of convenience.

That's fine...so long as it doesn't become a matter of performance.

> It's a SQLAlchemy artifact and it wouldn't hurt to keep them.

I have no idea whether that is true.  What causes the joins to happen?  Do we have to do some sort of explicit operation, or is it the case that, just because we declared the relationshi, a join is done on _every query_?

As I said, once we have a large number of datasets and a significant number of users, I expect that the joins will become costly, and we'll want to avoid such costs where they aren't needed.

I'm a big fan of code-as-documentation, but much less so when there is a runtime cost for it.  If we don't have a requirement for this relationship, we should consider omitting it until we have a need for it.